core/src/sumeragi/mod.rs

```diff
@@ -55,16 +56,31 @@ trait Consensus {
     ) -> Option<VersionedPendingBlock>;
 }
 
+/*
+The values in the following struct are not atomics because the code that
+operates on them assumes their values does not change during the course of
+the function.
+*/
+#[derive(Debug)]
+struct LastUpdateMetricsData {
+    block_height: u64,
+    metric_tx_amounts: f64,
+    metric_tx_amounts_counter: u64,
+}
+
 /// `Sumeragi` is the implementation of the consensus.
+#[allow(clippy::mutex_integer)]
```

Comment:
1. This should be enabled module-wide using `#![allow(…)`, given that you allow it almost everywhere it's used. 
2. This lint warns you very precisely about not using a `Mutex` wrapping an integer. I don't see an integer being wrapped, given that both `Metrics` and `LastUpdateMetricsData` is a `struct`. If this is a false-positive, reporting it would be nice.

Reply:
Oh yea sorry, I forgot to remove this one. Are there ways to detect unneeded allow statements?